src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SequenceEqualImmutableArray.cs

```diff
@@ -22,6 +22,11 @@ public SequenceEqualImmutableArray(ImmutableArray<T> array)
         {
         }
 
+        public SequenceEqualImmutableArray<T> Add(T value)
+        {
+            return new SequenceEqualImmutableArray<T>(Array.Add(value));
+        }
```

Comment:
This isn't an `Add()`. It is a `CopyAndExpand()`. Collections that have the work "immutable" in their name should not have operations that don't make that clear. Same for the `Insert()` below. These both should be updated because you can't have mutable verbs on an immutable collection.

Reply:
These methods exist on `ImmutableArray`, though, so my understanding was that if there's a modifying method on an "Immutable" type it copies and returns the new object. I'm fine renaming if you still prefer that though.

Reply:
Ugh. That design is terrible. Never mind, you are just following the existing pattern. Thanks for pointing this out.